compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/binary/KSTypeArgumentDescriptorImpl.kt

```diff
@@ -17,24 +17,34 @@
 
 package com.google.devtools.ksp.symbol.impl.binary
 
-import com.google.devtools.ksp.symbol.*
+import com.google.devtools.ksp.symbol.KSAnnotation
+import com.google.devtools.ksp.symbol.KSNode
+import com.google.devtools.ksp.symbol.KSTypeReference
+import com.google.devtools.ksp.symbol.Location
+import com.google.devtools.ksp.symbol.NonExistLocation
+import com.google.devtools.ksp.symbol.Origin
+import com.google.devtools.ksp.symbol.Variance
 import com.google.devtools.ksp.symbol.impl.KSObjectCache
 import com.google.devtools.ksp.symbol.impl.kotlin.IdKey
 import com.google.devtools.ksp.symbol.impl.kotlin.KSTypeArgumentImpl
 import org.jetbrains.kotlin.types.TypeProjection
 
-class KSTypeArgumentDescriptorImpl private constructor(val descriptor: TypeProjection, override val origin: Origin) :
-    KSTypeArgumentImpl() {
-    companion object : KSObjectCache<IdKey<Pair<TypeProjection, Origin>>, KSTypeArgumentDescriptorImpl>() {
-        fun getCached(descriptor: TypeProjection, origin: Origin) = cache.getOrPut(IdKey(Pair(descriptor, origin))) {
-            KSTypeArgumentDescriptorImpl(descriptor, origin)
-        }
+class KSTypeArgumentDescriptorImpl private constructor(
+    val descriptor: TypeProjection,
+    override val origin: Origin,
+    override val parent: KSNode?
+) : KSTypeArgumentImpl() {
+    companion object : KSObjectCache<IdKey<Triple<TypeProjection, Origin, KSNode?>>, KSTypeArgumentDescriptorImpl>() {
```

Comment:
Same here. However, you likely still need parent because a TypeProjection might be reused across different places.